packages/ember-meta/lib/meta.ts

```diff
@@ -440,82 +452,65 @@ export class Meta {
     method: Function | string,
     once: boolean
   ) {
-    if (this._listeners === undefined) {
-      this._listeners = [];
-    }
-    this._listeners.push(eventName, target, method, once);
+    this._addToListeners(eventName, target, method, once ? ListenerFlags.ONCE : ListenerFlags.NONE);
   }
 
-  _finalizeListeners() {
-    if (this._listenersFinalized) {
-      return;
-    }
-    if (this._listeners === undefined) {
-      this._listeners = [];
-    }
-    let pointer = this.parent;
-    while (pointer !== null) {
-      let listeners = pointer._listeners;
-      if (listeners !== undefined) {
-        this._listeners = this._listeners.concat(listeners);
-      }
-      if (pointer._listenersFinalized) {
-        break;
-      }
-      pointer = pointer.parent;
+  private _addToListeners(
+    eventName: string,
+    target: object | null,
+    method: Function | string,
+    flags: ListenerFlags
+  ) {
+    let { _listeners: listeners } = this;
+    if (listeners === undefined) {
+      listeners = this._listeners = [];
     }
-    this._listenersFinalized = true;
+    listeners.push(eventName, target, method, flags);
```

Comment:
should this dedupe entries? Specifically last write win, so that an `add` followed by `remove` doesn't leak the add?